gammapy/estimators/extension.py

```diff
@@ -0,0 +1,140 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+import astropy.units as u
+from astropy.coordinates import Angle
+from gammapy.datasets import Datasets
+from gammapy.estimators.parameter import ParameterEstimator
+from gammapy.modeling import Fit
+from gammapy.utils.pbar import progress_bar
+
+
+class ExtensionEstimator(ParameterEstimator):
+    """Source extension estimator.
+
+    Estimates source size for a given list of datasets and a given model.
+
+    The size can be estimated in various energy intervals.
+
+    Parameters
+    ----------
+    energy_edges : `~astropy.units.Quantity`
+        Edges of the energy intervals where the extension is estimated.
+    source : str or int
+        For which source in the model to compute the extension.
+    size_min : float
+        Minimum value for the size used for the fit statistic profile evaluation.
+    size_max : float
+        Maximum value for the size used for the fit statistic profile evaluation.
+    size_n_values : int
+        Number of size values used for the fit statistic profile.
+    size_values : `~astropy.coordinates.Angle`
+        Array of size values to be used for the fit statistic profile.
+    n_sigma : int
+        Number of sigma to use for asymmetric error computation. Default is 1.
+    n_sigma_ul : int
+        Number of sigma to use for upper limit computation. Default is 2.
+    selection_optional : list of str
+        Which additional quantities to estimate. Available options are:
+
+            * "all": all the optional steps are executed
+            * "errn-errp": estimate asymmetric errors on size.
+            * "ul": estimate upper limits.
+            * "scan": estimate fit statistic profiles.
+
+        Default is None so the optional steps are not executed.
+    fit : `Fit`
+        Fit instance specifying the backend and fit options.
+    reoptimize : bool
+        Re-optimize other free model parameters. Default is True.
+    """
+
+    tag = "ExtensionEstimator"
+
+    def __init__(
+        self,
+        energy_edges=[1, 10] * u.TeV,
```

Comment:
Maybe these default values should be removed... to be generic and to force the user to add edges